### ISSUE TITLE
Extend Metadata schema

### ISSUE THREAD

**Issue by depsimon** [sergiodxa/web-oidc#177]
It'd be great if we could easily extend/override the Metadata schema for the issuers.

I've tried a couple of OIDC configurations (ADB2C & Duende IdentityServer) and both have custom `claims_supported`, `grant_types_supported`, `response_types_supported` etc..

I haven't found an easy way to override that at the moment.

**Comment by JasonWeise** [comment]
I agree this is really needed as I don't seem to be able to authenticate with OneLogin due to Zod and schema matching issues.

**Comment by tonywoode** [comment]
my attempts to use this to add Microsoft OIDC SSO suffer the same problem here

**Comment by rivertam** [comment]
Does this basically mean these providers are simply not supported at all? Dreading recreating OIDC just because the zod schema doesn't match (although maybe the claims/grants need to be actually supported).
